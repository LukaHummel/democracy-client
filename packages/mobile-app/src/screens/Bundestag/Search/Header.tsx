import React, { useContext } from 'react';
import { SearchContext } from '../../../context/Search';
import { useMutation } from '@apollo/client';
import { FINISH_SEARCH } from './graphql/mutation/finishSearch';
import { SearchBar } from '@democracy-deutschland/ui';
import {
  FinishSearch,
  FinishSearchVariables,
} from './graphql/mutation/__generated__/FinishSearch';
import debounce from 'lodash.debounce';
import { Platform } from 'react-native';
import { styled } from '../../../styles';

const Wrapper = styled.View`
  background-color: ${({ theme }) => theme.oldColors.background.header};
`;

export const SearchHeader: React.FC = () => {
  const { setTerm, term, addToHistory } = useContext(SearchContext);
  const [executeFinishSearch] = useMutation<
    FinishSearch,
    FinishSearchVariables
  >(FINISH_SEARCH);

  const finishSearch = () => {
    console.log('finishSearch', term);
    addToHistory(term);
    executeFinishSearch({
      variables: {
        term,
      },
    });
  };

  // throttle to handle android endless changing error
  const onChangeText = (text: string) => {
    console.log(typeof term, term);
    if (typeof term === 'string') {
      console.log('onChangeText', text);
      Platform.OS === 'ios'
        ? setTerm(text)
        : debounce(() => setTerm(text), 300);
    }
  };
  return (
    <Wrapper>
      <SearchBar
        textInput={{
          placeholder: 'Suche',
          onSubmitEditing: finishSearch,
          onChangeText: onChangeText,
          value: term,
        }}
      />
      {/* <SearchBar2
        ref={searchBarRef}
        placeholder="Suche"
        text={term}
        onChangeText={onChangeText}
        onSearchButtonPress={finishSearch}
        showsCancelButton={false}
        showsCancelButtonWhileEditing={false}
        textFieldBackgroundColor="rgba(255,255,255,.5)"
        hideBackground
      /> */}
    </Wrapper>
  );
};
