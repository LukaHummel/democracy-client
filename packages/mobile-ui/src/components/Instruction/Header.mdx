---
name: Header
route: /
---

import { Playground, Props } from 'docz';
import Header from './Header';

# Button

<Props of={Button} />

## Basic usage

<Playground>
  <Header
    title="the title"
    description="the description"
    image={require('./assets/icon.logo.png')}
  />
</Playground>
