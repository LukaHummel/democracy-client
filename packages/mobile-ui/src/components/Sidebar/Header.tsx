import React from 'react';
import styled from 'styled-components/native';
import { HeadLogo } from './HeadLogo';
import { ActivityIndicator } from 'react-native';

const Container = styled.TouchableOpacity`
  flex-direction: row;
  padding-top: 16;
  padding-left: 16;
  padding-bottom: 8;
`;

const HeadTextWrapper = styled.View`
  justify-content: center;
`;

const HeadText = styled.Text`
  color: #fff;
  font-size: 17;
  padding-left: 16;
`;

interface Props {
  onPress: () => void;
  label: string;
}

export const Header: React.FC<Props> = ({ onPress, label }) => (
  <Container onPress={onPress}>
    {label !== 'verbindet…' && <HeadLogo />}
    {label === 'verbindet…' && <ActivityIndicator size="large" />}
    <HeadTextWrapper>
      <HeadText>{label}</HeadText>
    </HeadTextWrapper>
  </Container>
);
