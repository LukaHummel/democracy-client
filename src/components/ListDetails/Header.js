import React from 'react';
import styled from 'styled-components/native';
import PropTypes from 'prop-types';

import Votes from '../../components/Votes';
import Menu from '../../components/ListDetails/HeaderMenu';

const Wrapper = styled.View`
  padding-top: 27;
  padding-left: 27;
  padding-bottom: 5;
  border-bottom-width: 1;
  border-bottom-color: #c8c7cc;
`;

const HeaderMain = styled.View`
  flex-direction: row;
  justify-content: space-between;
  padding-bottom: 5;
`;

const Title = styled.Text.attrs({
  numberOfLines: 2,
})`
  flex: 1;
  font-size: 17;
  color: #030303;
  letter-spacing: -0.41;
`;

const Header = ({
  title, votes, commentsCount, places,
}) => (
  <Wrapper>
    <HeaderMain>
      <Title>{title}</Title>
      <Votes votes={votes} />
    </HeaderMain>
    <Menu commentsCount={commentsCount} places={places} />
  </Wrapper>
);

Header.defaultProps = {
  title: '',
  votes: 0,
  commentsCount: 0,
  places: 0,
};

Header.propTypes = {
  title: PropTypes.string,
  votes: PropTypes.number,
  commentsCount: PropTypes.number,
  places: PropTypes.number,
};

export default Header;
