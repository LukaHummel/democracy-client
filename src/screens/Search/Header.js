import React, { Component } from "react";
import { Platform } from "react-native";
import styled from "styled-components/native";
import Ionicons from "react-native-vector-icons/Ionicons";
import PropTypes from "prop-types";
import { Navigator } from "react-native-navigation";

const Wrapper = styled.View`
  flex: 1;
  flex-direction: row;
  align-items: center;
  margin-right: ${Platform.OS === "ios" ? 0 : 16};
`;

const Icons = styled(Ionicons.Button).attrs({
  color: "#fff",
  size: 30,
  backgroundColor: "transparent"
})``;

const SearchInput = styled.TextInput.attrs({
  clearButtonMode: "always",
  autoFocus: true,
  placeholderTextColor:
    Platform.OS === "ios" ? "#7a797b" : "rgba(255, 255, 255, 0.38)",
  underlineColorAndroid: "transparent",
  selectionColor: Platform.OS === "ios" ? "#000" : "#fff",
  returnKeyType: "search"
})`
  flex: 1;
  background-color: ${Platform.OS === "ios"
    ? "rgba(255, 255, 255, 0.8)"
    : "transparent"};
  border-radius: 5.5;
  font-size: ${Platform.OS === "ios" ? 14 : 20};
  height: ${Platform.OS === "ios" ? 28 : "auto"};
  padding-horizontal: ${Platform.OS === "ios" ? 6 : 0};
  color: ${Platform.OS === "ios" ? "#000" : "#fff"};
`;

const SearchBackButtonAndroid = styled(Icons).attrs({
  name: "md-arrow-back"
})`
  padding-right: 16;
`;

const SearchClearButtonAndroid = styled(Icons).attrs({
  name: "md-close"
})`
  display: ${({ visible }) => (visible ? "flex" : "none")};
  padding-left: 16;
`;

const SearchBackTextIos = styled.Button.attrs({
  color: "#fff"
})`
  padding-left: 8;
  font-size: 17;
`;

class Header extends Component {
  state = {
    searchContent: ""
  };

  clickBack = () => {
    const { navigator } = this.props;
    navigator.pop();
  };

  render() {
    const { searchContent } = this.state;
    return (
      <Wrapper>
        {Platform.OS === "android" && (
          <SearchBackButtonAndroid onPress={this.clickBack} />
        )}
        <SearchInput
          placeholder="Suche"
          onChangeText={text => this.setState({ searchContent: text })}
          value={searchContent}
        />

        {Platform.OS === "android" && (
          <SearchClearButtonAndroid
            visible={searchContent}
            onPress={() => this.setState({ searchContent: "" })}
          />
        )}
        {Platform.OS === "ios" && (
          <SearchBackTextIos title="ZurÃ¼ck" onPress={this.clickBack} />
        )}
      </Wrapper>
    );
  }
}

Header.propTypes = {
  navigator: PropTypes.instanceOf(Navigator)
};

Header.defaultProps = {
  navigator: undefined
};

export default Header;
