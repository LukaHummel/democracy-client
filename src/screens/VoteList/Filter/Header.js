import React, { Component } from "react";
import { Platform } from "react-native";
import styled from "styled-components/native";
import { Navigation } from "react-native-navigation";

const Wrapper = styled.View`
  flex: 1;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
`;

const HeaderElement = styled.View`
  flex: 1;
  padding-right: ${Platform.OS === "android" ? 11 : 0};
`;

const TitleText = styled.Text.attrs({})`
  font-size: 17;
  color: #fff;
  font-weight: bold;
`;

const Button = styled.TouchableOpacity``;

const ButtonText = styled.Text`
  font-size: 17;
  color: #fff;
`;

class Header extends Component {
  onSave = async () => {
    await this.props.onSave();
    this.clickBack();
  };

  clickBack = () => {
    Navigation.dismissModal({
      animationType: "slide-down" // 'none' / 'slide-down' , dismiss animation for the modal (optional, default 'slide-down')
    });
  };

  render() {
    return (
      <Wrapper>
        {Platform.OS === "ios" && (
          <HeaderElement style={{ alignItems: "flex-start" }}>
            <Button onPress={this.clickBack}>
              <ButtonText>Zur√ºck</ButtonText>
            </Button>
          </HeaderElement>
        )}
        <HeaderElement style={{ alignItems: "center" }}>
          <TitleText>FILTER</TitleText>
        </HeaderElement>
        <HeaderElement style={{ alignItems: "flex-end" }}>
          <Button onPress={this.onSave}>
            <ButtonText>Speichern</ButtonText>
          </Button>
        </HeaderElement>
      </Wrapper>
    );
  }
}

Header.propTypes = {};

Header.defaultProps = {};

export default Header;
