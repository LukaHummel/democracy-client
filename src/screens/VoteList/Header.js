import React from "react";
import styled from "styled-components/native";

const Wrapper = styled.View`
  background-color: red;
`;

const Title = styled.Text``;

const Header = () => (
  <Wrapper>
    <Title>Der Titel</Title>
  </Wrapper>
);

export default Header;
