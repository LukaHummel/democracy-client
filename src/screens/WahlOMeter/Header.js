import React from 'react';
import styled from 'styled-components/native';

const Wrapper = styled.View`
  flex-direction: row;
`;

const TimeSelect = styled.View`
  flex: 1;
`;

const ProcedureCount = styled.View``;

const ProcedureCountText = styled.Text`
  font-size: 15;
  color: #6d6d72;
  font-weight: 500;
`;

const WomHeader = ({ totalProcedures, votedProceduresCount }) => (
  <Wrapper>
    <TimeSelect />
    <ProcedureCount>
      <ProcedureCountText>
        {votedProceduresCount}/{totalProcedures}
      </ProcedureCountText>
    </ProcedureCount>
  </Wrapper>
);

export default WomHeader;
