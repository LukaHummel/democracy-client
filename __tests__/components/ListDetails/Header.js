import 'react-native';
import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import Header from '../../../src/components/ListDetails/Header';

describe('test components/List/Row', () => {
  it('render row', () => {
    const wrapper = renderer
      .create(<Header title="the title" votes={31523} commentsCount={84} places={3} />)
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
