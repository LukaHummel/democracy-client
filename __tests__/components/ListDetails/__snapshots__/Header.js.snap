// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test components/List/Row render row 1`] = `
<View
  style={
    Array [
      Object {
        "borderBottomColor": "#c8c7cc",
        "borderBottomWidth": 1,
        "paddingBottom": 5,
        "paddingLeft": 27,
        "paddingTop": 27,
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingBottom": 5,
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={2}
      style={
        Array [
          Object {
            "color": "#030303",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "fontSize": 17,
            "letterSpacing": -0.41,
          },
          undefined,
        ]
      }
    >
      the title
    </Text>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "width": 48,
          },
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "grey",
              "fontSize": 12,
            },
            Array [
              Object {
                "fontSize": 45,
                "marginBottom": -15,
                "marginTop": -10,
              },
              undefined,
            ],
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
      <Text
        accessible={true}
        adjustsFontSizeToFit={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#8f8e94",
              "fontSize": 18,
              "letterSpacing": -0.29,
            },
            undefined,
          ]
        }
      >
        31523
      </Text>
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "width": 50,
          },
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#4a4a4a",
              "fontSize": 24,
            },
            Array [
              Object {},
              undefined,
            ],
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#4a4a4a",
              "fontSize": 12,
              "letterSpacing": -0.26,
            },
            undefined,
          ]
        }
      >
        3
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "width": 50,
          },
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#4a4a4a",
              "fontSize": 24,
            },
            Array [
              Object {},
              undefined,
            ],
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#4a4a4a",
              "fontSize": 12,
              "letterSpacing": -0.26,
            },
            undefined,
          ]
        }
      >
        84
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "width": 50,
          },
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#44db5e",
              "fontSize": 12,
              "letterSpacing": -0.26,
            },
            undefined,
          ]
        }
      >
        3:21
      </Text>
    </View>
  </View>
</View>
`;
